bool got_r
string resource_name
string object_name
int32 id
